src/visibility_tracker.rs

```diff
@@ -15,15 +15,15 @@ pub(crate) struct VisibilityTracker<'a> {
 
 impl<'a> VisibilityTracker<'a> {
     pub(crate) fn from_crate(crate_: &'a Crate) -> Self {
-        let visible_parent_ids = compute_parent_ids_for_public_items(crate_)
-            .into_iter()
-            .map(|(key, values)| {
-                // Ensure a consistent order, since queries can observe this order directly.
-                let mut values: Vec<_> = values.into_iter().collect();
-                values.sort_unstable_by_key(|x| &x.0);
-                (key, values)
-            })
-            .collect();
+        let mut visible_parent_ids = compute_parent_ids_for_public_items(crate_);
+
+        let iter = visible_parent_ids.iter_mut();
+
+        // Sort and deduplicate parent ids
+        iter.for_each(|(_id, parent_ids)| {
+            parent_ids.sort_unstable_by_key(|id| id.0.as_str());
+            parent_ids.dedup_by_key(|id| id.0.as_str());
+        });
```

Comment:
A bit of personal preference, if you don't mind indulging me: I find it easier to understand iterator `.for_each()` when the base iterator is constructed inline instead of constructed separately and assigned to a variable.
```suggestion
        // Sort and deduplicate parent ids.
        // This ensures a consistent order, since queries can observe the order directly.
        visible_parent_ids.iter_mut().for_each(|(_id, parent_ids)| {
            parent_ids.sort_unstable_by_key(|id| id.0.as_str());
            parent_ids.dedup_by_key(|id| id.0.as_str());
        });
```

Reply:
Fixed!